### PR TITLE
docs: add swingset docs page on typography

### PR DESCRIPTION
**Pull request by zchsh**
🎟️ [Asana Task](https://app.asana.com/0/1100423001970639/1200604505419290/f)
🔍 [Preview Link](https://react-components-git-zsadd-typography-docs-to-7368db-hashicorp.vercel.app/docs/typography)

---

This PR adds documentation on our typefaces and on our global `g-type-*` utility classes. 

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [x] Add Asana and Preview links above.
- [x] Conduct thorough self-review.
- [x] Add or update tests as appropriate.
- [x] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [x] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [x] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: docs/typography.mdx [hunk 1]
```diff
@@ -0,0 +1,156 @@
+---
+name: 'Typography'
+---
+
+<!-- workaround to drop swingset's styling of first h1 -->
+
+<h1 />
```
Comment (zchsh) [inline docs/typography.mdx:7]: Copied this pattern from the `color.mdx` page as I had the same intent for the `h1`.

File: docs/typography.mdx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+# Typography
```
Comment (zchsh) [inline docs/typography.mdx:9]: Much of this content is ported over from `web-components` (it joins the former [typography](https://web-components-hashicorp.vercel.app/?path=/docs/base-theme-typography--page) and [g-type-*](https://web-components-hashicorp.vercel.app/?path=/story/packages-global-styles-components-g-type--demo) pages).

File: docs/typography.mdx [hunk 3]
```diff
@@ -0,0 +1,156 @@
+---
+name: 'Typography'
```
Comment (zchsh) [inline docs/typography.mdx:2]: I'm having an odd issue where I can see `/docs/typography` locally, but can't see it in the deploy preview. @BRKalow did you run into any similar issues deploying the `/docs/color` page?

Comment (BRKalow) [reply docs/typography.mdx:2]: That's odd...not that I remember. Let me take a look!

**Comment by BRKalow** [comment]
@zchsh Tried to push up an empty commit to see if it'll show up after a rebuild. I did a production build locally and the page _did_ show up.

**Comment by zchsh** [comment]
Seems to be working now - https://react-components-git-zsadd-typography-docs-to-7368db-hashicorp.vercel.app/docs/typography 👍 👍 👍

**Review by jescalan — APPROVED** [review]
This is so great!

**Pull request merged**